src/main/java/org/humancellatlas/ingest/project/ProjectService.java

```diff
@@ -175,4 +182,33 @@ private ProjectBag gather(Project project) {
         return new ProjectBag(projects, envelopes);
     }
 
+    public Page<Project> filterProjects(String search, String wranglingState, String wrangler, Pageable pageable) {
+        List<Criteria> criterias = new ArrayList<>();
+        criterias.add(Criteria.where("isUpdate").is(false));
+        Query query = new Query();
+
+        if (wranglingState != null) {
```

Comment:
You could use Optional to remove a lot of boilerplate code here.

Reply:
Hmm, so I tried doing something like:

```
 Optional<String> opt = Optional.of(searchFilter.getWranglingState());
     opt.isPresent((wranglingState -> {
      criterias.add(Criteria.where("wranglingState").is((Object) wranglingState)
  });
```

and got the compiler error:

```
Required: Object
Provided: lambda parameter
```

I haven't been able to find out a fix for this yet, any suggestions?

Reply:
`isPresent` is a parameter-less method that returns a boolean. You should use `ifPresent`.